src/app/components/audio-recordings/pages/download/download.component.ts

```diff
@@ -0,0 +1,389 @@
+  implements OnInit
+{
+  public fields: FormlyFieldConfig[];
+  public model: Model = {};
+  public models: ResolvedModelList;
+  public href = "";
+  public failure: boolean;
+  public contactUs = contactUsMenuItem;
+  public profile = myAccountMenuItem;
+
+  public constructor(
+    private config: ConfigService,
+    private route: ActivatedRoute,
+    private recordingsApi: AudioRecordingsService
+  ) {
+    super();
+  }
+
+  public ngOnInit(): void {
+    const models = retrieveResolvers(new PageInfo(this.route.snapshot.data));
+    if (!hasResolvedSuccessfully(models)) {
+      this.failure = true;
+      return;
+    }
+    this.models = models;
+    this.updateHref();
+
+    const hideProjects = this.config.settings.hideProjects;
+    this.fields = [];
+
+    if (!hideProjects) {
+      if (this.project) {
+        this.fields.push(this.getProjectFormInput());
+      } else {
+        // TODO Future work, read project list and make typeahead
+      }
+    }
+
+    if (this.region) {
+      this.fields.push(this.getRegionFormInput(hideProjects));
+    } else {
+      // TODO Future work, read regions/sites list and make typeahead
+    }
+
+    if (this.site) {
+      this.fields.push(this.getSiteFormInput(hideProjects));
+    } else {
+      // TODO Future work, read sites list and make typeahead
+    }
+
+    this.fields.push(...this.dateSelectionRange());
+  }
+
+  public get project(): Project {
+    return this.models[projectKey] as Project;
+  }
+  public get region(): Region {
+    return this.models[regionKey] as Region;
+  }
+  public get site(): Site {
+    return this.models[siteKey] as Site;
+  }
+
+  private getProjectFormInput(): FormlyFieldConfig {
+    return {
+      key: "projectId",
+      type: "select",
+      defaultValue: this.project.id,
+      templateOptions: {
+        label: "Project",
+        // TODO Allow user to change selected site
+        disabled: true,
+        options: [{ value: this.project.id, label: this.project.name }],
+      },
+    };
+  }
+
+  private getRegionFormInput(hideProject: boolean): FormlyFieldConfig {
+    return {
+      key: "regionId",
+      type: "select",
+      defaultValue: this.region.id,
+      templateOptions: {
+        label: hideProject ? "Project" : "Site",
+        // TODO Allow user to change selected site
+        disabled: true,
+        options: [{ value: this.region.id, label: this.region.name }],
+      },
+    };
+  }
+
+  private getSiteFormInput(hideProject: boolean): FormlyFieldConfig {
+    return {
+      key: "siteId",
+      type: "select",
+      defaultValue: this.site.id,
+      templateOptions: {
+        label: hideProject ? "Site" : "Point",
+        // TODO Allow user to change selected site
+        disabled: true,
+        options: [{ value: this.site.id, label: this.site.name }],
+      },
+    };
+  }
+
+  private dateSelectionRange(): FormlyFieldConfig[] {
+    return [
+      {
+        key: "recordingStartedBefore",
```

Comment:
```suggestion
        key: "recordingStartedAfter",
```

though personall I'd choose start and end